crypto/asn1/x_int64.c

```diff
@@ -111,12 +117,23 @@ static int uint32_i2c(ASN1_VALUE **pval, unsigned char *cont, int *putype,
         && utmp == 0)
         return -1;
     if ((it->size & INTxx_FLAG_SIGNED) == INTxx_FLAG_SIGNED
-        && (int32_t)utmp < 0)
+        && (int32_t)utmp < 0) {
+        /* i2c_uint64_int() assumes positive values */
+        utmp = 0 - utmp;
         neg = 1;
+    }
 
     return i2c_uint64_int(cont, (uint64_t)utmp, neg);
 }
 
+/*
+ * Absolute value of INT32_MIN: we can't just use -INT32_MIN as it produces
+ * overflow warnings.
+ */
+
+#define ABS_INT32_MIN \
+    ((uint32_t)INT32_MAX + (uint32_t)(-(INT32_MIN + INT32_MAX)))
```

Comment:
Have a look for `ABS_INT64_MIN`, that's where I copied the formula from.  Yours does seem simpler enough.